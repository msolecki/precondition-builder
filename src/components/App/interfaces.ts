export interface LocationInterface {
    radius: number | null;
    latLng: string | null;
}

enum OperatingSystemInterface {
    iOS = 'iOS',
    Android = 'Android',
}

export interface NuggetConditionInterface {
    id: number;
    condition: 'read';
    value: boolean;
}

export interface ConditionInterface {
    logged: boolean | null;
    location: LocationInterface | null;
    nuggets: NuggetConditionInterface[] | null;
}
