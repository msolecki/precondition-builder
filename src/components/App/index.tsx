import React, {ChangeEvent, ReactElement} from 'react'
import {Container} from '@material-ui/core'
import AppBar from '@material-ui/core/AppBar'
import Tabs from '@material-ui/core/Tabs'
import Tab from '@material-ui/core/Tab'
import Typography from '@material-ui/core/Typography'
import AdvancedTab from '../Tabs/AdvancedTab'
import BasicTab from '../Tabs/BasicTab'
import {ConditionInterface} from './interfaces'

export interface StateInterface {
    conditions: ConditionInterface[];
    tabNumber: number;
}

class App extends React.Component<{}, StateInterface> {
    public constructor(props: {}) {
        super(props)

        this.state = {
            conditions: [],
            tabNumber: 0
        }
    }

    private changeTab = (event: ChangeEvent<{}>, newNumber: number): void => {
        this.setState({tabNumber: newNumber})
    }

    public handleAddCondition = (newCondition: ConditionInterface): void => {
        const conditions = this.state.conditions
        conditions.push(newCondition)

        this.setState({conditions})
    }

    public render(): ReactElement {
        const {tabNumber, conditions} = this.state

        return (
            <Container style={{width: '50%'}}>
                <AppBar position="static">
                    <Tabs value={tabNumber} onChange={this.changeTab} variant="fullWidth">
                        <Tab label="Basic"/>
                        <Tab label="Advanced"/>
                    </Tabs>
                </AppBar>
                <Typography component="div" style={{paddingTop: 25}}>
                    {tabNumber === 0 && <BasicTab handleAddCondition={this.handleAddCondition} conditions={conditions}/>}
                    {tabNumber === 1 && <AdvancedTab data={conditions}/>}
                </Typography>
            </Container>
        )
    }
}

export default App
