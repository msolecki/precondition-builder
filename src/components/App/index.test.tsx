import * as React from 'react'
import {ReactTestRenderer} from 'react-test-renderer'
import {createShallow} from '@material-ui/core/test-utils'

import App, {AppProps} from './index'

const component = (props: AppProps) => <App {...props}/>

describe('<App />', () => {
    let shallow

    beforeAll(() => {
        shallow = createShallow()
    })

    test('empty snapshot', () => {
        const wrapper: ReactTestRenderer = shallow(component({
            htmlNuggetIds: ['1'],
        }))

        expect(wrapper).toMatchSnapshot()
    })
})
