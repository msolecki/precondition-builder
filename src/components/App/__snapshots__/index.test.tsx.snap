// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> empty snapshot 1`] = `
<WithStyles(ForwardRef(Container))
  style={
    Object {
      "width": "50%",
    }
  }
>
  <WithStyles(ForwardRef(AppBar))
    position="static"
  >
    <WithStyles(ForwardRef(Tabs))
      onChange={[Function]}
      value={0}
      variant="fullWidth"
    >
      <WithStyles(ForwardRef(Tab))
        label="Basic"
      />
      <WithStyles(ForwardRef(Tab))
        label="Advanced"
      />
    </WithStyles(ForwardRef(Tabs))>
  </WithStyles(ForwardRef(AppBar))>
  <WithStyles(ForwardRef(Typography))
    component="div"
    style={
      Object {
        "paddingTop": 25,
      }
    }
  >
    <BasicTab
      activated={false}
      conditions={Array []}
      handleActivated={[Function]}
      handleAddCondition={[Function]}
      htmlNuggetIds={
        Array [
          "1",
        ]
      }
    />
  </WithStyles(ForwardRef(Typography))>
</WithStyles(ForwardRef(Container))>
`;
